packages/utils/src/schema/sanitizeDataForNewSchema.ts

```diff
@@ -0,0 +1,208 @@
+      let newKeyedSchema: S = get(newSchema, [PROPERTIES_KEY, key], {});
+      // Resolve the refs if they exist
+      if (has(oldKeyedSchema, REF_KEY)) {
+        oldKeyedSchema = retrieveSchema<T, S, F>(
+          validator,
+          oldKeyedSchema,
+          rootSchema,
+          formValue
+        );
+      }
+      if (has(newKeyedSchema, REF_KEY)) {
+        newKeyedSchema = retrieveSchema<T, S, F>(
+          validator,
+          newKeyedSchema,
+          rootSchema,
+          formValue
+        );
+      }
+      // Now get types and see if they are the same
+      const oldSchemaTypeForKey = get(oldKeyedSchema, "type");
+      const newSchemaTypeForKey = get(newKeyedSchema, "type");
+      // Check if the old option has the same key with the same type
+      if (!oldSchemaTypeForKey || oldSchemaTypeForKey === newSchemaTypeForKey) {
+        if (has(removeOldSchemaData, key)) {
+          // SIDE-EFFECT: remove the undefined value for a key that has the same type between the old and new schemas
+          delete removeOldSchemaData[key];
+        }
+        // If it is an object, we'll recurse and store the resulting sanitized data for the key
+        if (
+          newSchemaTypeForKey === "object" ||
+          (newSchemaTypeForKey === "array" && Array.isArray(formValue))
+        ) {
+          // SIDE-EFFECT: process the new schema type of object recursively to save iterations
+          const itemData = sanitizeDataForNewSchema<T, S, F>(
+            validator,
+            rootSchema,
+            newKeyedSchema,
+            oldKeyedSchema,
+            formValue
+          );
+          if (itemData !== undefined || newSchemaTypeForKey === "array") {
+            // only put undefined values for the array type and not the object type
+            nestedData[key] = itemData;
+          }
+        } else {
+          // Ok, the non-object types match, let's make sure that a default or a const of a different value is replaced
+          // with the new default or const
+          const newOptionDefault = get(newKeyedSchema, ["default"]);
+          const newOptionConst = get(newKeyedSchema, ["const"]);
+          if (
+            (newOptionDefault && newOptionDefault !== formValue) ||
+            (newOptionConst && newOptionConst !== formValue)
+          ) {
+            removeOldSchemaData[key] = newOptionDefault;
+          }
+        }
+      }
+    });
+
+    newFormData = {
+      ...data,
+      ...removeOldSchemaData,
+      ...nestedData,
+    };
+    // First apply removing the old schema data, then apply the nested data, then apply the old data keys to keep
+  } else if (
+    get(oldSchema, "type") === "array" &&
+    get(newSchema, "type") === "array" &&
+    Array.isArray(data)
+  ) {
+    let oldSchemaItems = get(oldSchema, "items");
+    let newSchemaItems = get(newSchema, "items");
+    // If any of the array types `items` are either booleans or arrays then we'll just drop the data
```

Comment:
Would it be reasonable to keep the data if they are both `true`?

Reply:
Probably... I wonder what kind of data would be here if they were both true.